app/models/user.rb

```diff
@@ -0,0 +1,22 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+
+  validates :name, presence: true,
+                   length: {maximum: Settings.valid.name_max_len}
+
+  validates :email, presence: true,
+                    format: {with: Settings.regex.email_regex},
+                    uniqueness: {case_sensitive: false},
+                    length: {maximum: Settings.valid.email_max_len}
+
+  has_secure_password
+
+  validates :password, presence: true,
+                       length: {minimum: Settings.valid.password_min_len}
+
+  private
+
+  def downcase_email
+    email.downcase!
```

Comment:
so sánh method `downcase` vs `downcase!`

Reply:
downcase và downcase! đếu dùng để chuổi toàn bộ chuỗi về chữ cài thường, nhưng trong downcase! sẽ sửa trực tiếp lên chuỗi đó còn downcase sẽ trả về một chuỗi mới
VD: 
a = "HELLO"
a.downcase! 
print a # prints "hello", a = "hello"

a = "HELLO"
print a.downcase => "hello", a = "HELLO"